### PR TITLE
Better handle hex values as bytes

### PR DESCRIPTION
**Pull request by stevenroose**
Basically replaces all remaining strings that are actually hex representations of bytes into `Vec<u8>`.

### CODE REVIEW

File: client/src/client.rs [hunk 1]
```diff
@@ -99,7 +104,7 @@ fn handle_defaults<'a, 'b>(
     let required_num = args.len() - defaults.len();
 
     if let Some(i) = first_non_null_optional_idx {
-        &args[..i+1]
+        &args[..i + 1]
```
Comment (dpc) [inline client/src/client.rs]: Did `rustfmt` do that? It looks weird that the space is between operator that has precedence. Also, can't we just `[..=i]`? Is the rustc version requirement too low?

File: client/src/client.rs [hunk 2]
```diff
@@ -366,44 +364,38 @@ pub trait RpcApi: Sized {
         self.call("signrawtransaction", handle_defaults(&mut args, &defaults))
     }
 
-    fn sign_raw_transaction_with_key(
+    fn sign_raw_transaction_with_key<B: AsRef<[u8]>>(
         &self,
-        tx: json::HexBytes,
+        tx: B,
         privkeys: &[&str],
```
Comment (sgeisler) [inline client/src/client.rs]: Might be out of scope, but matches the spirit of the PR: why not also make `privkeys` a slice of `PrivateKey`s?

File: client/src/client.rs (hunk 1 above)
Comment (stevenroose) [reply client/src/client.rs]: Yeah somehow my Vim plugin `:RustFmt` doesnt format the same as `cargo fmt`. I'm still not sure how that came. These are some tiny nits from `cargo fmt`.

File: client/src/client.rs (hunk 2 above)
Comment (stevenroose) [reply client/src/client.rs]: Yeah this was done before `PrivateKey` had serde. I think now it has. There should be a todo on some occurrence of that. Might PR that at some point :)

**Pull request merged**